File: Mage/CoreData/ImageryLayer.m
//
//  ImageryLayer+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by Dan Barela on 10/1/19.
//  Copyright © 2019 National Geospatial-Intelligence Agency. All rights reserved.
//
//

#import "ImageryLayer.h"

@implementation ImageryLayer

- (id) populateObjectFromJson: (NSDictionary *) json withEventId: (NSNumber *) eventId {
    [self setRemoteId:[json objectForKey:@"id"]];
    [self setName:[json objectForKey:@"name"]];
    [self setType:[json objectForKey:@"type"]];
    [self setUrl:[json objectForKey:@"url"]];
    [self setEventId:eventId];
    [self setFormat:[json objectForKey:@"format"]];
    [self setOptions:[json objectForKey:@"wms"]];
    [self setIsSecure:[[json objectForKey:@"url"] hasPrefix:@"https"]];
    return self;
}

@end

File: Mage/CoreData/ImageryLayer.swift
//
//  ImageryLayer+CoreDataClass.m
//  mage-ios-sdk
//
//  Created by Dan Barela on 10/1/19.
//  Copyright © 2019 National Geospatial-Intelligence Agency. All rights reserved.
//
//

import Foundation
import CoreData
import Layer
import Persistence

extension ImageryLayer {
    @objc public override func populate(_ json: [AnyHashable : Any], eventId: NSNumber) {
        super.populate(json, eventId: eventId)
        self.format = json[LayerKey.format.key] as? String
        self.options = json[LayerKey.wms.key] as? [AnyHashable : Any]
        self.isSecure = self.url?.hasPrefix("https") ?? false
    }
}
